src/main/java/cuchaz/enigma/EnigmaServices.java

```diff
@@ -4,18 +4,19 @@
 import cuchaz.enigma.api.service.EnigmaService;
 import cuchaz.enigma.api.service.EnigmaServiceType;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
 
 public final class EnigmaServices {
-	private final ImmutableMap<EnigmaServiceType<?>, EnigmaService> services;
+	private final ImmutableMap<EnigmaServiceType<?>, List<EnigmaService>> services;
```

Comment:
This could use `ImmutableMultimap`

Reply:
Does gson parse ImmutableMultimap? This is currently auto-parsed by GSON

Reply:
Ah, I see. Probably worth testing, can't find anything online